### PR TITLE
docs: updated multiselect docs with select all feature details and usage

### LINKED ISSUE #17689: [story]: Add the `selectAll` prop usage for multiselect
Need to write the documentation how the `selectAll` prop can be used with the multiselect.

### PR DESCRIPTION
**Pull request by Gururajj77**
Closes #17689 

Added documentation about select all overview and usage  in the  Multiselect's storybook over page

#### Changelog

**New**

- select all feature documentation

#### Testing / Reviewing

go to deploy preview, check the multiselect component overview page, scroll down and see the select all documentation in the page

### CODE REVIEW

File: packages/react/src/components/MultiSelect/MultiSelect.mdx [hunk 1]
```diff
@@ -71,6 +71,30 @@ For more information, checkout the Downshift prop
...
+The MultiSelect component now includes a "Select All" feature, allowing users to
+easily select or deselect all options with a single action.
+
+To implement the "Select All" feature in your MultiSelect component:
+
+Add a new option to your options array with these properties:
+
+- id: A unique identifier (e.g., 'select-all')
+- text: The label for the "Select All" option
+- isSelectAll: Set to true
```
Comment (kennylam) [inline packages/react/src/components/MultiSelect/MultiSelect.mdx]:
```suggestion
To implement the "Select All" feature in your MultiSelect component,
add a new option to your options array with these properties:

- `id`: A unique identifier (e.g., 'select-all')
- `text`: The label for the option
- `isSelectAll`: Set to `true`
```

Looks good! Just a small suggestion to clean up the instructions.

**Review by kennylam — APPROVED** [review]

File: packages/react/src/components/MultiSelect/MultiSelect.mdx [hunk 2]
```diff
@@ -71,6 +71,29 @@ For more information, checkout the Downshift prop
 
 <ArgTypes />
 
+## New "Select All" Feature
```
Comment (alisonjoseph) [inline packages/react/src/components/MultiSelect/MultiSelect.mdx]:
```suggestion
### Select all
```

File: packages/react/src/components/MultiSelect/MultiSelect.mdx [hunk 3]
```diff
@@ -71,6 +71,29 @@ For more information, checkout the Downshift prop
...
+
+The MultiSelect component now includes a "Select All" feature, allowing users to
```
Comment (alisonjoseph) [inline packages/react/src/components/MultiSelect/MultiSelect.mdx]:
```suggestion
The MultiSelect component includes a "Select All" feature, allowing users to
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
I would change the wording a little, also might be nice to add to the nav at the top of the storybook docs also, could easily be missed otherwise at the bottom of the page.

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**